src/hooks/useStore.js

```diff
@@ -1,33 +1,30 @@
-import {nanoid} from 'nanoid';
+import axios from 'axios';
 import create from 'zustand';
 
-const useStore = create(set => {
+const useStore = create((set, get) => {
 	return {
-		items: [
-			{item: 'Zelt', id: nanoid(), isChecked: false},
-			{item: 'Grill', id: nanoid(), isChecked: false},
-			{item: 'Wein', id: nanoid(), isChecked: false},
-			{item: 'noch mehr Wein', id: nanoid(), isChecked: false},
-		],
+		items: [],
+		getItems: async () => {
+			console.log('fetching items');
+			const response = await axios.get('/api/connect');
+			const result = await response.data;
 
-		addItem: item => {
-			set(state => {
-				return {items: [...state.items, {item, id: nanoid(), isChecked: false}]};
-			});
+			set(() => ({items: result}));
 		},
-		deleteItem: id => {
-			set(state => {
-				return {items: state.items.filter(item => item.id !== id)};
-			});
-		},
-		checkItem: id => {
-			set(state => {
-				return {
-					items: state.items.map(item =>
-						item.id === id ? {...item, isChecked: !item.isChecked} : item
-					),
-				};
-			});
+
+		addItems: async item => {
+			console.log('adding items');
+			try {
+				const result = await fetch('/api/connect', {
+					method: 'POST',
+					body: JSON.stringify({item}),
+				});
+				get().getItems();
+
+				return await result.json();
```

Comment:
Here is a big opportunity for an optimization. Instead of another fetch with the `get().getItems()` you could get back the list with the answer of the `fetch('/api/connect)', function. And save it with the normal zustand `set` function.

```suggestion
				const result = await fetch('/api/connect', {
					method: 'POST',
					body: JSON.stringify({item}),
				});
				const result = await result.json()
				set(() => ({items: result.data}));
				return await result.json();
```

And see changes in [src/pages/api/connect.js](https://github.com/JoaLehma/capstone-project/pull/9/files#diff-098c5a2496d59bbedb5cb1e46ffb2b1339e321eb0b3066cfc106dd1643ed1e61)

Reply:
thx for your feedback. i think there is a mistake. both const are named "result" (row 18 and 22)?

i have tried this:
`	const response = await fetch('/api/connect', {
					method: 'POST',
					body: JSON.stringify({item}),
				});
				const result = await response.json();
				set(() => ({items: result.data}));
				return await result.json();`

and also updated the api as suggested below.
now, after submitting a new item, this happes:
"Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'map')" in my StyledForm component.

Reply:
ok, konnte das problem mit hilfe von dominik beheben :) das hier war die lösung:
![Bildschirmfoto 2022-08-26 um 13 28 39](https://user-images.githubusercontent.com/107816788/186894006-20a2008c-7b78-4b4d-9908-5f808071b6ba.png)